dags/analyzer_helper/tests/integration/test_extract_raw_info.py

```diff
@@ -0,0 +1,124 @@
+        cls.extractor.close()
+        cls.driver.close()
+
+    def test_fetch_post_details(self):
+        post_details = self.extractor.fetch_post_details()
+        self.assertEqual(len(post_details), 2)
+        author_names = [post['author_name'] for post in post_details]
+        self.assertIn('User One', author_names)
+        self.assertIn('User Two', author_names)
+
+        for post in post_details:
+            self.assertIn('post_id', post)
+            self.assertIn('author_id', post)
+            self.assertIn('created_at', post)
+            self.assertIn('author_name', post)
+            self.assertIn('reactions', post)
+            self.assertIn('replied_post_id', post)
+            self.assertIn('topic_id', post)
+
+    def test_fetch_post_categories(self):
+        post_details = self.extractor.fetch_post_details()
+        post_ids = [post["post_id"] for post in post_details]
+        post_categories = self.extractor.fetch_post_categories(post_ids)
+        self.assertEqual(len(post_categories), 2)
+
+        for category in post_categories:
+            self.assertIn('post_id', category)
+            self.assertIn('category_id', category)
+
+    def test_fetch_raw_data(self):
+        combined_data = self.extractor.fetch_raw_data()
+        self.assertEqual(len(combined_data), 2)
+        author_names_categories = [(post['author_name'], post['category_id']) for post in combined_data]
+        self.assertIn(('User One', 'category1'), author_names_categories)
+        self.assertIn(('User Two', 'category1'), author_names_categories)
+
+        for post in combined_data:
+            self.assertIn('post_id', post)
+            self.assertIn('author_id', post)
+            self.assertIn('created_at', post)
+            self.assertIn('author_name', post)
+            self.assertIn('reactions', post)
+            self.assertIn('replied_post_id', post)
+            self.assertIn('topic_id', post)
+            self.assertIn('category_id', post)
+            
+    def test_extract_with_recompute(self):
+        self.rawmemberactivities_collection.delete_many({})
+
+        period = datetime.now()
+        data = self.extractor.extract(period, recompute=True)
+
+        # Check if data is fetched from the Neo4j database without date filtering
+        self.assertEqual(len(data), 2)
+        author_names = [post['author_name'] for post in data]
+        self.assertIn('User One', author_names)
+        self.assertIn('User Two', author_names)
+
+    def test_extract_without_recompute_no_latest_activity(self):
+        self.rawmemberactivities_collection.delete_many({})
+        
+        result = self.extractor.extract(period=datetime(2023, 1, 1), recompute=False)
+        self.assertGreater(len(result), 0)
+    
+    def test_extract_without_recompute_latest_activity_before_period(self):
+        self.rawmemberactivities_collection.delete_many({})
+        self.rawmemberactivities_collection.insert_one({'date': '2022-12-31T00:00:00Z'})
```

Comment:
Add The whole document with its right structure. it's not right to just to add one field as the real data won't be like this.
I've seen this kind of assertion on your previous PRs, please make sure to follow the rules of adding the exact right data and not a part of data.